File: app/src/main/java/com/example/githubapp/presentation/common/SchedulersProvider.java
package com.example.githubapp.presentation.common;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class SchedulersProvider {

  public Scheduler io() {
    return Schedulers.io();
  }

  public Scheduler computation() {
    return Schedulers.computation();
  }

  public Scheduler ui() {
    return AndroidSchedulers.mainThread();
  }
}

File: app/src/main/java/com/example/githubapp/presentation/common/SchedulersProvider.kt
package com.example.githubapp.presentation.common

import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
import io.reactivex.rxjava3.core.Scheduler
import io.reactivex.rxjava3.schedulers.Schedulers

class SchedulersProvider {
    fun io(): Scheduler {
        return Schedulers.io()
    }

    fun computation(): Scheduler {
        return Schedulers.computation()
    }

    fun ui(): Scheduler {
        return AndroidSchedulers.mainThread()
    }
}